envoyauth/response.go

```diff
@@ -0,0 +1,254 @@
+package envoyauth
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	ext_core_v3 "github.com/envoyproxy/go-control-plane/envoy/config/core/v3"
+	ext_type_v3 "github.com/envoyproxy/go-control-plane/envoy/type/v3"
+	"github.com/open-policy-agent/opa-envoy-plugin/internal/util"
+	"github.com/open-policy-agent/opa/metrics"
+)
+
+//EvalResult - Captures the result from evaluating a query against an input
+type EvalResult struct {
+	Revision   string // Deprecated: Use `revisions` instead.
+	Revisions  map[string]string
+	DecisionID string
+	TxnID      uint64
+	Decision   interface{}
+	Metrics    metrics.Metrics
+}
+
+//StopFunc should be called as soon as the evaluation is finished
```

Comment:
[nit] I'd probably add a whitespace after every `//`. But it's just another these silly little code-style comments, feel free to ignore.

Reply:
Changed